atd-vze/src/views/Users/Users.js

```diff
@@ -83,8 +87,57 @@ const Users = () => {
       });
   }, [token, page, perPage]);
 
+  useEffect(() => {
```

Comment:
It would be cool to explore wrapping the new state and useEffect in a custom hook to keep the new functionality in a logical chunk, but your documentation about what is going on here makes it straightforward to see what is happening! 🙏